File: Sep/Day8/convert-integer-to-the-sum-of-two-no-zero-integers.js
//  Time: O(N * LogN)  ::  Space: O(1)
class Solution {
    getNoZeroIntegers(n) {
        let a = 1, b = n - a;
        while ((a.toString().includes("0") || b.toString().includes("0")) && a <= b) {
            a++;
            b--;
        }
        return [a, b];
    }
}

File: Sep/Day8/convert-integer-to-the-sum-of-two-no-zero-integers.cc
//  Time: O(N * LogN)  ::  Space: O(1)
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> getNoZeroIntegers(int n) {
        int a = 1, b = n - a;
        while ((to_string(a).find('0') != string::npos || to_string(b).find('0') != string::npos) && a <= b) {
            a++;
            b--;
        }
        return {a, b};
    }
};
